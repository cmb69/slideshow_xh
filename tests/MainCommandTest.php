<?php

/**
 * Copyright 2021 Christoph M. Becker
 *
 * This file is part of Slideshow_XH.
 *
 * Slideshow_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Slideshow_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Slideshow_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Slideshow;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\View;

class MainCommandTest extends TestCase
{
    public function testRendersView()
    {
        $imageRepo = $this->createStub(ImageRepo::class);
        $imageRepo->method('findAll')
            ->willReturn([new Image("pics/foo.jpg"), new Image("pics/bar.jpg")]);
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["slideshow"]);
        $sut = new MainCommand($imageRepo, $view);
        Approvals::verifyHtml($sut("pics"));
    }
}
