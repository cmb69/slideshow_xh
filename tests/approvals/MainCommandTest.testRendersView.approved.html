
<!-- Slideshow_XH slideshow -->
<script type="module" src="./plugins/slideshow/slideshow.min.js"></script>
<div class="slideshow" style="position: relative; width: 100%; overflow: hidden"
     data-effect="fade" data-easing="easeInOut"
     data-delay="0" data-pause="2000" data-duration="1000">
  <picture style="position: static; display: block; z-index: 1; width: 100%">
    <source srcset="pics/foo.webp" type="image/webp">
    <img src="pics/foo.jpg" alt="foo" loading="eager">
  </picture>
  <picture style="position: absolute; display: none; width: 100%">
    <source srcset="pics/bar.avif" type="image/avif">
    <img src="pics/bar.jpg" alt="bar" loading="lazy">
  </picture>
</div>
